### ISSUE TITLE
[Bug] Reconnect seems not working

### ISSUE THREAD

**Issue by shevernitskiy** [harmonyland/harmony#342]
**What's the bug?**
Seems like `reconnect` client method not working.

**How do we reproduce it?**
Run this:

```ts
const discord = new Client();

setTimeout(async () => {
  await discord.reconnect();
  console.log("reconnected"); // never reached
}, 20 * 1000);

discord.on("reconnect", () => {
   console.log("reconnected"); // never triggered as well
});

discord.connect(config.discord_token, Intents.All)
```

**What should have happened?**
Client should reinitialized connection.

**What is actually happening?**
Bot still working as usual, but without reconnect.

**What versions you're using?**
 - OS:  win 11
 - Deno: 1.31.1
 - Harmony: 2.8.0

**Do you have anything to tell us more about the bug?**
I'm trying to perform reconnect, cause in some random case bot hangs and stop receiving any updates. There are no errors, but also no reaction to commands or other events.

**Comment by Helloyunho** [comment]
it's been about 8 months since this issue opened... sorry about that 😅
seems like the reconnect function falls into a loop, I'll fix this ASAP

**Comment by Helloyunho** [comment]
fixed with commit e3e3c73056b1980ee214d7047aa2d104c515f737
